homeassistant/components/twinkly/light.py

```diff
@@ -0,0 +1,224 @@
+"""The Twinkly light component."""
+
+import asyncio
+import logging
+from typing import Any, Dict, Optional
+
+from aiohttp import ClientError
+import twinkly_client
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    SUPPORT_BRIGHTNESS,
+    LightEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import (
+    ATTR_HOST,
+    CONF_ENTRY_HOST,
+    CONF_ENTRY_ID,
+    CONF_ENTRY_MODEL,
+    CONF_ENTRY_NAME,
+    DEV_MODEL,
+    DEV_NAME,
+    DOMAIN,
+    HIDDEN_DEV_VALUES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, config_entry: ConfigEntry, async_add_entities
+) -> None:
+    """Setups an entity from a config entry (UI config flow)."""
+    uuid = config_entry.data[CONF_ENTRY_ID]
+    host = config_entry.data[CONF_ENTRY_HOST]
+    name = config_entry.data[CONF_ENTRY_NAME]
+    model = config_entry.data[CONF_ENTRY_MODEL]
+
+    client = twinkly_client.TwinklyClient(host, async_get_clientsession(hass))
```

Comment:
If you setup the `TwinklyClient` in `__init__`, you can pull it here from `hass.data`